商品情報編集機能

naoya1215 opened the pull request:
#What
商品情報編集機能

#Why
商品情報編集機能実装のため

#提出物
ログイン状態の出品者は、商品情報編集ページに遷移できる動画
https://gyazo.com/d4611854106ccee2a7bd192e0ea51024

必要な情報を適切に入力して「更新する」ボタンを押すと、商品の情報を編集できる動画
https://gyazo.com/dfd608d73712343faaad49f8e7406884

入力に問題がある状態で「変更する」ボタンが押された場合、情報は保存されず、編集ページに戻りエラーメッセージが表示される動画
https://gyazo.com/79e1f814b3e5a2c9bf987e3022076c03

何も編集せずに「更新する」ボタンを押しても、画像無しの商品にならない動画
https://gyazo.com/9354c605471aa0c4d5820ae57278c81c

ログイン状態の場合でも、URLを直接入力して自身が出品していない商品の商品情報編集ページへ遷移しようとすると、商品の販売状況に関わらずトップページに遷移する動画
※補足：hoge2がhogeの出品した編集画面へ移動しようとしている
https://gyazo.com/39c057df2f06e202e543d8ed8c4c090e

ログイン状態の場合でも、URLを直接入力して自身が出品した売却済み商品の商品情報編集ページへ遷移しようとすると、トップページに遷移する動画（現段階で商品購入機能の実装が済んでいる場合）
商品購入機能未実装

ログアウト状態の場合は、URLを直接入力して商品情報編集ページへ遷移しようとすると、商品の販売状況に関わらずログインページに遷移する動画
※補足：ログアウト状態でhogeの出品した編集画面へ移動しようとしている
https://gyazo.com/4182c9341b037a102686feb0b65954ef

商品名やカテゴリーの情報など、すでに登録されている商品情報は商品情報編集画面を開いた時点で表示される動画（商品画像・販売手数料・販売利益に関しては、表示されない状態で良い）
https://gyazo.com/81616a8890f5c8a730a530447f473c9c

naoya1215 committed:
編集機能の実装

必要な情報を適切に入力して「変更する」ボタンを押すと、商品情報（商品画像・商品名・商品の状態など）を編集できること。

naoya1215 committed:
redirect_toの設定

何も編集せずに「変更する」ボタンを押しても、画像無しの商品にならないこと。

naoya1215 committed:
戻るのパスを指定

 ページ下部の「もどる」ボタンを押すと、編集途中の情報は破棄され、商品詳細表示ページに遷移すること。

naoya1215 committed:
投稿者以外が編集ページへ遷移することを防ぐ

ログイン状態の場合でも、自身が出品していない商品の商品情報編集ページへ遷移しようとすると、商品の販売状況に関わらずトップページに遷移すること。

naoya1215 committed:
コメントの修正と追記

不要なコメントと躓いたところはコメントを残す

naoya1215 committed:
編集ページのエラー表示

編集時に入力エラーが発生した場合、対応するエラーが出るようにインスタンス変数(@item)にする

Akinori0123 commented on app/controllers/items_controller.rb:
```diff
@@ -24,10 +25,34 @@ def show # 詳細ページへ
 
+  def edit #編集ページへ
+    # new.html.erbからクリックされた情報をパラメータとして受け取り、edit.html.erbで編集ページを表示
+    @item = Item.find(params[:id])
```
```
@item = Item.find(params[:id])
```
コントローラ内に同じ@itemの定義が複数ありますので、privateメソッドにまとめてbefore_actionで呼び出しましょう。
DRY原則に則ったリファクタリングのためです。

Akinori0123 reviewed (commented):
レビュー依頼いただきありがとうございます！
コミット内容と添付いただいたgyazoの方確認いたしました。
コメントいたしましたのでご確認の程よろしくお願いいたします。

naoya1215 committed:
リファクタリング

メンターアドバイスにより、修正
@item = Item.find(params[:id])
コントローラ内に同じ@itemの定義が複数ありますので、privateメソッドにまとめてbefore_actionで呼び出しましょう。
DRY原則に則ったリファクタリングのためです。

naoya1215 committed:
Rubocopの実行

bundle exec rubocop -aのコマンドを入力し、自動整形

naoya1215 commented:
早速コメントいただき有難うございます。
DRY原則、頭から離れておりました、修正しましたので確認をお願いします。

【再提出】

ログイン状態の出品者は、商品情報編集ページに遷移できる動画
https://gyazo.com/1ddb982913bcdf9949a1b9a39db49d0e

必要な情報を適切に入力して「更新する」ボタンを押すと、商品の情報を編集できる動画
https://gyazo.com/34001df76eb83d30449cb1db64599be3

入力に問題がある状態で「変更する」ボタンが押された場合、情報は保存されず、編集ページに戻りエラーメッセージが表示される動画
https://gyazo.com/092884f4ae5ac4ae9ca63e7df1427e4b

何も編集せずに「更新する」ボタンを押しても、画像無しの商品にならない動画
https://gyazo.com/f25854f436103ab09b3831adab90b418

ログイン状態の場合でも、URLを直接入力して自身が出品していない商品の商品情報編集ページへ遷移しようとすると、商品の販売状況に関わらずトップページに遷移する動画
※補足：hoge2がhogeの出品した編集画面へ移動しようとしている
https://gyazo.com/375b50f81f3f76b314d9c112c9ebbbbd

ログイン状態の場合でも、URLを直接入力して自身が出品した売却済み商品の商品情報編集ページへ遷移しようとすると、トップページに遷移する動画（現段階で商品購入機能の実装が済んでいる場合）
商品購入機能未実装

ログアウト状態の場合は、URLを直接入力して商品情報編集ページへ遷移しようとすると、商品の販売状況に関わらずログインページに遷移する動画
※補足：ログアウト状態でhogeの出品した編集画面へ移動しようとしている
https://gyazo.com/b7557ee390b7bb151ab90ae9c4bff445

商品名やカテゴリーの情報など、すでに登録されている商品情報は商品情報編集画面を開いた時点で表示される動画（商品画像・販売手数料・販売利益に関しては、表示されない状態で良い）
https://gyazo.com/1ddb982913bcdf9949a1b9a39db49d0e

TakumaKawashima reviewed (commented):
レビュー依頼ありがとうございます。

修正点確認いたしました。備忘録としてコメントも良く記載出来ていますね。
LGTMとさせていただきます。実装お疲れさまでした。

マージ後、ローカルのmasterブランチに忘れずにプルしましょう。
その後、新しいブランチを作成し次の実装に取り組んでいただけますと幸いです。
引き続き頑張っていきましょう。
